### PR TITLE
Replace immutable with immer

### PR DESCRIPTION
**Pull request by stereosaurus**
# Pull Request Template

Please Verify the following before submitting a pull request:
* If there is a related Github issue, the pull request name includes the issue number
* Unit tests are added/fixed if applicable. All unit tests must pass
* Code is formatted correctly, via prettier (npm run prettier) or terraform (terraform fmt)
* Commit message following the Conventional Commits standard is provided below

> You can delete everything above when you are ready to submit your pull request

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -85,6 +85,7 @@
 		"express-session": "^1.17.1",
 		"glob": "^7.1.4",
 		"helmet": "^4.2.0",
+		"immer": "^9.0.12",
 		"immutable": "^4.0.0-rc.12",
```
Comment (Robert-W) [inline package.json]: Lets remove immutable since its no longer being used

Comment (stereosaurus) [reply package.json]: Oh right. I told myself I would and totally forgot. Good catch

File: src/client/routing/SecureRoute.js [hunk 2]
```diff
@@ -16,9 +16,9 @@ export default function SecureRoute(props) {
 			path={path}
 			exact={exact}
 			render={(routerProps) => {
-				let roles = user.getIn(['data', 'roles']);
+				let roles = user.data.roles
 				// Only render the target page if the user has the required roles
-				if (requiredRoles.isSubset(roles) || roles.includes('admin')) {
+				if (requiredRoles.every((role) => roles.includes(role))) {
```
Comment (Robert-W) [inline src/client/routing/SecureRoute.js]: Hmm, we may need to update the logic in every to also include the `role.includes('admin')`, something like:
```
if (roles.includes('admin') || requiredRoles.every((role) => roles.includes(role))) {
```

but let me know what you think. My though is that be default, we allow admins to see every page.  Currently if we configured the following routes:

```
{
  requiredRoles: [],
  name: 'Home'
},
{
  requiredRoles: ['manager'],
  name: 'Dashboard'
},
```

Everyone and admin could see Home, but only managers would see the Dashboard, the admin would not be able to see it unless he also had the manager role.

Comment (stereosaurus) [reply src/client/routing/SecureRoute.js]: Makes sense. I agree. I'll put that fix in thanks.

**Review by Robert-W — APPROVED** [review]

**Pull request merged**